signer/jwt.go

```diff
@@ -4,25 +4,25 @@ import (
 	"crypto"
 	"crypto/ecdsa"
 	"crypto/rsa"
+	"crypto/x509"
 	"encoding/base64"
+	"encoding/json"
 	"fmt"
 
 	"github.com/golang-jwt/jwt/v4"
 )
 
 // signJWT signs the given payload and headers using the given signing method and signature provider
-func signJWT(payload, headers map[string]interface{}, method jwt.SigningMethod, sigPro SignatureProvider) (string, error) {
-	var claims jwt.MapClaims = payload
-	token := &jwt.Token{
-		Header: headers,
-		Claims: claims,
-	}
+func signJWT(payload []byte, headers map[string]interface{}, sigPro SignatureProvider) (string, []*x509.Certificate, error) {
```

Comment:
Are we using any functionality from golang-jwt during signing, if not, can we implement signing wholly in JWS layer?

Reply:
Done.